core/state_prefetcher.go

```diff
@@ -153,5 +153,8 @@ func precacheTransaction(msg types.Message, config *params.ChainConfig, gaspool
 	// Update the evm with the new transaction context.
 	evm.Reset(NewEVMTxContext(msg), statedb)
 	// Add addresses to access list if applicable
-	ApplyMessage(evm, msg, gaspool)
+	if _, err := ApplyMessage(evm, msg, gaspool); err == nil {
+		statedb.Finalise(true)
```

Comment:
What's the difference if we call tire_prefetcher here? Not sure is there any other side effects.

Reply:
Actually, we should do Finalise() here, to flush DirtyObjects into PendingObjects and clear the journal.
It can help to make the concurrent StatePrefetch's behavior more similar to the main process.
It is ok, should have no side effect.